src/grt/src/fastroute/src/utility.cpp

```diff
@@ -237,6 +237,10 @@ void FastRouteCore::fillVIA()
           newCNT++;
         }
 
+        if(routeLen <= 0) {
```

Comment:
An edge representing a via can have len == 0, right? My comment about the error message was for the case where both len > 0 and a routeLen < 0, which should never happen.

Reply:
Nevermind, I see that this check happens after checking for len > 0.

Reply:
yes, but in this case len > 0 already true, and if an edge has routeLen <= 0 meens that this edge was never routed, witch should be an error in this stage of the fastroute, that is because len and routeLen represent different information, len is the manhattan distance between the nodes of the edge, where routeLen is the size of the route's X, Y and L vectors. The case where routeLen == 0 should also be an error, and looking at the clang-tidy warning it would also access garbage values.